frontend/src/metabase/list-view/components/ListView/ColumnValue.tsx

```diff
@@ -1,91 +1,286 @@
+            <Stack mr="0.25rem">
+              <Box
+                style={{
+                  borderRadius: "50%",
+                  width: "0.5rem",
+                  height: "0.5rem",
+                  backgroundColor: getCategoryColor(rawValue, column.name),
+                }}
+              />
+            </Stack>
+          }
+        >
           {value}
-        </Text>
-      </Flex>
-    );
-  }
+        </Badge>
+      );
+    case TYPE.State:
+      return (
+        <Badge
+          className={styles.badge}
+          size="lg"
+          variant="outline"
+          style={{
+            background: "var(--mb-color-bg-white)",
+          }}
+        >
+          {value}
+        </Badge>
+      );
+    case TYPE.Name:
+    case TYPE.Title:
+    case TYPE.Product:
+    case TYPE.Source:
+      return (
+        <Ellipsified
+          size="sm"
+          truncate
+          fw="bold"
+          style={style}
+          c={style?.color || "text-primary"}
+        >
+          {value}
+        </Ellipsified>
+      );
+    case TYPE.Email:
+    case TYPE.URL:
+      return (
+        <Ellipsified size="sm" fw="bold" style={style} tooltip={rawValue}>
+          {value}
+        </Ellipsified>
+      );
+    case TYPE.Quantity:
+    case TYPE.Score: {
+      return (
+        <Flex direction="row" align="center" gap="sm">
+          <MiniBarCell
+            rowIndex={0}
+            columnId={column.name}
+            value={Number(rawValue)}
+            barWidth="4rem"
+            barHeight="0.25rem"
+            extent={columnExtent}
+            columnSettings={columnSettings || DEFAULT_COLUMN_SETTINGS}
+            style={{
+              paddingInline: 0,
+              marginLeft: 0,
+              width: "auto",
+              border: "none",
+            }}
+          />
+          <Text fw="bold">{value}</Text>
+        </Flex>
+      );
+    }
+    case TYPE.Percentage: {
+      return (
+        <Badge
+          size="lg"
+          className={styles.badge}
+          variant="outline"
+          style={{
+            background: "var(--mb-color-bg-white)",
+          }}
+        >
+          <Text fw="bold">
+            {String(value).slice(0, -1)}
+            <span
+              style={{
+                color: "var(--mb-color-text-tertiary)",
+                paddingLeft: "0.25rem",
+              }}
+            >
+              %
+            </span>
+          </Text>
+        </Badge>
+      );
+    }
+    case TYPE.Currency: {
+      const options = settings.column?.(column) || {};
+      const formattedValue = formatNumber(Number(rawValue), options);
 
-  if (shouldGetCategoryStyling && rawValue != null && rawValue !== "") {
-    return (
-      <Box
-        className={styles.categoryValue}
-        style={{ backgroundColor: getCategoryColor(rawValue, column.name) }}
-      >
-        <Ellipsified fw="bold" size="sm" c="text-secondary">
+      if (
+        options.currency_style === "symbol" &&
```

Comment:
There are 4 possible `currency_style` values, we should account for all.

Reply:
For other style types we'll just render the value as a regular bold text. I think we should postpone other improvements on this particular feature, because we don't currently have a proper tooling to detach currency label from the value amount (which is required to style it properly).